include/maxscale/buffer.h

```diff
@@ -423,4 +423,14 @@ extern void dprintAllBuffers(void *pdcb);
  */
 void gwbuf_hexdump(GWBUF* buffer);
 
+/**
+ * Return pointer of the byte at offset from start of chained buffer
+ *
+ * @param buffer  one or more chained buffer
+ * @param offset  Offset into the buffer
+ * @return  if total buffer length is bigger than offset then return
+ *      the offset byte pointer, otherwise return null
+ */
+extern uint8_t *gwbuf_byte_point(GWBUF* buffer, size_t offset);
```

Comment:
I think `gwbuf_data_pointer` would be a better name as it returns a pointer to the data in the GWBUF.

Also add a note that the returned pointer is not guaranteed to point to a contiguous segment of memory. In other words, it is only safe to modify the first byte of data that the return value points to.